pkg/protocompat/time.go

```diff
@@ -52,6 +52,40 @@ func CompareTimestamps(t1 *gogoTimestamp.Timestamp, t2 *gogoTimestamp.Timestamp)
 	return t1.Compare(t2)
 }
 
+// CompareTimestampToTime compares a proto timestamp to a time.
+// The return value is:
+// * -1 if the proto timestamp is before the time
+// * 0 if both represent the same time
+// * 1 if the proto timestamp is after the time
+func CompareTimestampToTime(t1 *gogoTimestamp.Timestamp, t2 *time.Time) int {
+	if t1 == nil && t2 == nil {
+		return 0
+	}
+	if t1 == nil {
+		return -1
+	}
+	if t2 == nil {
+		return 1
+	}
+	utcTime := t2.UTC()
```

Comment:
Can we utilize some convert function and then just compare the same types? i.e. (if we are using Golang 1.20, we could convert `t1` to `time.Time` and use `Compare()` function)